Framework/Algorithms/src/InterpolatingRebin.cpp

```diff
@@ -200,12 +201,9 @@ void InterpolatingRebin::outputYandEValues(
 *  @throw invalid_argument if any output x-values are outside the range of input
 *x-values
 **/
-void InterpolatingRebin::cubicInterpolation(const HistogramData::BinEdges &xOld,
-                                            const MantidVec &yOld,
-                                            const MantidVec &eOld,
-                                            const HistogramData::BinEdges &xNew,
-                                            MantidVec &yNew,
-                                            MantidVec &eNew) const {
+void InterpolatingRebin::cubicInterpolation(
+    const BinEdges &xOld, const HistogramY &yOld, const HistogramE &eOld,
+    const BinEdges &xNew, HistogramY &yNew, HistogramE &eNew) const {
```

Comment:
I would consider passing `Histogram` as arguments, instead of all the individual X, Y, and E vectors. That should also make things cleaner at the call site. Maybe `Histogram cubicInterpolation(const Histogram &histogram, const BinEdges &xNew)`.

Note that you can then also remove some of the lengths checks below, since `Histogram` guarantees this (if `xMode` is `BinEdges`, but you can make sure it is at the workspace level).